src/Connector.js

```diff
@@ -347,7 +355,12 @@ class Connector {
 
   async publishData(id, dataList) {
     const promises = dataList.map(async (data) => {
-      await this.client.publish(`/${id}/${data.sensorId}/attrs/value`, data.value.toString());
+      let value = data.value.toString();
+      if (isBase64(data.value)) {
+        value = Buffer.from(data.value, 'base64').toString('utf-8');
+      }
+
+      await this.client.publish(`/${id}/${data.sensorId}/attrs/value`, value);
```

Comment:
If using an await statement here the process will wait for the promise be resolved for each iteration of `map` function.

Besides that, the return of map will be an array of undefined. 
Please see the reference of map function:
https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
```suggestion
      return this.client.publish(`/${id}/${data.sensorId}/attrs/value`, value);
```

Reply:
Changes made!

Still, I wonder if functions such as `removeDeviceFromOrion()`,  `removeDeviceFromIoTAgent()` and `subscribeToEntities()` will have the same issues under `Promise.all(promises)` -- given that they have no return statement (inside `Promise.all`) and either subscribe or make requests.

Reply:
> About what you commented [here](https://github.com/CESARBR/knot-fog-connector-fiware/pull/10#discussion_r350387075) @lcbm :
> Yes, these functions have the same error. Please add a commit correcting them. Nice found!

@ramonhpr, made the requested changes! :innocent: 
Could you take a closer look in `subscribeToEntities()`?  :eyes: 
Thanks!!

Reply:
@lcbm I've resolved this comment. But I added new ones in the commit related to  the subscribe